packages/foundry/contracts/hooks/NftCheckHook.sol

```diff
@@ -0,0 +1,323 @@
+interface ICustomNFT is IERC721 {
+    struct NFTData {
+        string status;
+        address linkedToken;
+        string[] linkedTokenInterfaces;
+        bool locked;
+        bool paused;
+    }
+
+    function getNftData(uint256 tokenId) external view returns (NFTData memory);
+}
+
+contract NftCheckHook is BaseHooks, VaultGuard, Ownable {
+    using FixedPoint for uint256;
+
+    address public nftContract;
+    uint256 public nftId;
+    address public firstDepositor;
+    uint256 public initialToken1Amount;
+    uint256 public initialToken2Amount;
+    uint256 public redeemRatio;
+    address public poolAddress;
+    address private linkedTokenAddress;
+    address private token;
+    TokenConfig[] private tokenConfigs;
+
+    uint64 public exitFeePercentage;
+    uint64 public constant MAX_EXIT_FEE_PERCENTAGE = 10e16;
+
+    error DoesNotOwnRequiredNFT(address hook, address nftContract, uint256 nftId);
+    error LinkedTokenNotInPool(address linkedToken);
+    error InsufficientLiquidityToRemove(address user, uint256 currentAmount, uint256 initialAmount);
+    error InsufficientStableForSettlement(uint256 required, uint256 available);
+    error ExitFeeAboveLimit(uint256 feePercentage, uint256 limit);
+    error PoolDoesNotSupportDonation();
+
+    event NftCheckHookRegistered(address indexed hooksContract, address indexed pool);
+    event ExitFeeCharged(address indexed pool, IERC20 indexed token, uint256 feeAmount);
+    event ExitFeePercentageChanged(address indexed hookContract, uint256 exitFeePercentage);
+    event NftContractUpdated(address indexed oldContract, address indexed newContract);
+    event NftIdUpdated(uint256 oldId, uint256 newId);
+    event InitialLiquidityRecorded(address indexed user, uint256 token1Amount, uint256 token2Amount);
+    event LiquiditySettled(uint256 totalEscrowedAmount, address indexed originalDepositor);
+    event Redeemed(address indexed user, uint256 poolTokenAmount, uint256 stableTokenAmount);
+
+    constructor(IVault vault, address _nftContract, uint256 _nftId, address _linkedTokenAddress, address _token) VaultGuard(vault) Ownable(msg.sender) {
+        nftContract = _nftContract;
+        nftId = _nftId;
+        linkedTokenAddress = _linkedTokenAddress;
+        token =_token;
+    }
+
+    function onRegister(
+        address,
+        address pool,
+        TokenConfig[] memory _tokenConfigs,
+        LiquidityManagement calldata liquidityManagement
+    ) public override onlyVault returns (bool) {
+        if (liquidityManagement.enableDonation == false) {
+            revert PoolDoesNotSupportDonation();
+        }
+
+        tokenConfigs = _tokenConfigs;
+
+        // save pool address for later use
+        poolAddress = pool;
+
+        // Record the initial liquidity amounts for use in limiting the position from the depositor prematurely
+        recordInitialLiquidity(tokenConfigs[0].token.balanceOf(pool), tokenConfigs[1].token.balanceOf(pool));
```

Comment:
What if the pool has more than two tokens. Looks like you should prevent this.

Reply:
Liquidity is only added to the pool on initialize, at this point, the balances will always be 0.